srcopsmetrics/github_knowledge_store.py

```diff
@@ -0,0 +1,186 @@
+                time.sleep(wait_time)
+
+            if idx % 10 == 0:
+                _LOGGER.info("[ API requests remaining: %d ]" % remaining)
+
+            _LOGGER.info("Analysing %s no. %d/%d" %
+                         (self.entity_type, idx, len(self.new_entities)))
+
+            self.store_method(entity, self.accumulator)
+        return self.accumulator
+
+    def save_knowledge(self, file_path: Path, data: Dict[str, Any]):
+        """Save collected knowledge as json.
+
+        The saved json contains one dictionary with single key 'results'
+        under which the knowledge is stored.
+
+        Arguments:
+            file_path {Path} -- where the knowledge should be saved
+            data {Dict[str, Any]} -- collected knowledge. Should be json compatible
+        """
+        results = {"results": data}
+
+        _LOGGER.info("Saving knowledge file %s of size %d" %
+                    (os.path.basename(file_path), len(data)))
+
+        if not self.is_local:
+            ceph_filename = os.path.relpath(file_path).replace("./", "")
+            s3 = self.get_ceph_store()
+            s3.store_document(results, ceph_filename)
+            _LOGGER.info("Saved on CEPH at %s%s%s" %
+                        (s3.bucket, s3.prefix, ceph_filename))
+        else:
+            with open(file_path, "w") as f:
+                json.dump(results, f)
+            _LOGGER.info("Saved locally at %s" % file_path)
+
+    def get_ceph_store(self) -> CephStore:
+        """Establish a connection to the CEPH."""
+        s3 = CephStore(prefix=self._PREFIX, host=self._HOST, bucket=self._BUCKET)
+        s3.connect()
+        return s3
+
+    def load_previous_knowledge(
+        self, project_name: str, file_path: Path, knowledge_type: str,
+    ) -> Dict[str, Any]:
+        """Load previously collected repo knowledge. If a repo was not inspected before, create its directory.
+
+        Arguments:
+            repo_path {Path} -- path of the inspected github repository
+
+        Returns:
+            Dict[str, Any] -- previusly collected knowledge.
+                            Empty dict if the knowledge does not exist.
+
+        """
+        filename = self._FILENAME_ENTITY[knowledge_type]
+
+        if file_path is None:
+            pwd = Path.cwd().joinpath("./srcopsmetrics/bot_knowledge")
+            project_path = pwd.joinpath("./" + project_name)
+            file_path = project_path.joinpath("./" + filename + ".json")
+        
+        results = self.load_remotely(file_path) if not self.is_local else self.load_locally(file_path)
+
+        if results is None:
+            _LOGGER.info("No previous knowledge found for %s" % project_name)
+            results = {}
+            return results
+
+        _LOGGER.info("Found previous knowledge for %s with %d entities of type %s" % (
+            project_name, len(results), knowledge_type))
+
+        return results
+
+    @staticmethod
```

Comment:
what is the reason behind the static method?

Reply:
See comment below :)